stata_kernel/poor_mans_magic.py

```diff
@@ -0,0 +1,130 @@
+import argparse
+import regex
+
+magic_regex = regex.compile(
+    r'\A%(?<magic>.+?)(?<code>\s+.*)?\Z',
```

Comment:
`magic` and `code` are named capture groups? That you fill in using `stata_magics`?

Reply:
`magic` captures the word afer % if % is the first character in the code block. `code` captures all the things after. `stata_magics` is a dictionary that contains all the magics implemented. I wasn's sure what a nice way to implent magics was, but that is what I came uo with. Hance the `hack-ish` caveat.